models/project_board.go

```diff
@@ -96,6 +103,11 @@ func createBoardsForProjectsType(sess *xorm.Session, project *Project) error {
 // NewProjectBoard adds a new project board to a given project
 func NewProjectBoard(board *ProjectBoard) error {
 	_, err := x.Insert(board)
+	if board.Color != "" {
```

Comment:
Should check this before inserting the board into the database.
Maybe this should be moved into the router because validation of other models is made in the router and not in the model too.